requirements.txt

```diff
@@ -1,15 +1,15 @@
-numpy
-accelerate>=0.20.3
-packaging
-transformers>=4.34.1
-torch
-aim==3.17.5
-sentencepiece
-tokenizers>=0.13.3
-tqdm
-trl
-ninja
-peft>=0.8.0
-datasets>=2.15.0
-fire
+numpy>=1.26.4,<2.0
+accelerate>=0.20.3,<1.0
+transformers>=4.34.1,<5.0
+torch>=2.2.0,<3.0
+aim>=3.17.5,<4.0
+sentencepiece>=0.1.99,<2.0
+tokenizers>=0.13.3,<1.0
+tqdm>=4.66.2,<5.0
+trl>=0.7.10,<1.0
```

Comment:
for libraries that are not yet `1.0`, I think it is not guaranteed that there won't be any API breaking change between subversions, so it is probably better to be more conservative about their versions in range.

Reply:
applies less for `accelerate` though, since that library just never went `1.0` but is very popular, so unlikely to make such changes now without big splash.